agreementbot/agreementworker.go

```diff
@@ -331,15 +331,73 @@ func (b *BaseAgreementWorker) AgreementLockManager() *AgreementLockManager {
+        _, nodePolicy, err := compcheck.GetNodePolicy(nodePolicyHandler, wi.Device.Id, msgPrinter)
+
+        // If a deployment policy is being used, set wi.ProducerPolicy to the node policy
+        if wi.ConsumerPolicy.PatternId == "" {
+                // non pattern case
+
+                if err != nil {
+                        glog.Errorf(BAWlogstring(workerId, fmt.Sprintf("%v", err)))
+                        return
+                } else if nodePolicy == nil {
+                        glog.Warning(BAWlogstring(workerId, fmt.Sprintf("Cannot find node policy for this node %v.", wi.Device.Id)))
+                        return
+                } else {
+                        if glog.V(5) {glog.Infof(BAWlogstring(workerId, fmt.Sprintf("retrieved node policy: %v", nodePolicy)))}
+                        wi.ProducerPolicy = *nodePolicy
+                }
+
+                // If a deployment policy is being used and multiple service versions are possible, do an initial check of just the policy constraints of the deployment policy
+                // with the node properties to see if those match before we get too far invested in checking matches of all the different service versions.
+                // In the case were have thousands of deployment policies, this can avoid lots of calls to check and create workload_usages in the DB if there isn't a match at this level
+                //
+                // If the node policy has constraints, we can't do the check without the service properties so we need to find the workload.
+                // Also, if there is only 1 workload possibility, skip the check here and just do the check with the service included to avoid checking twice
+                // for the case where the node does match the policy
+                if len(nodePolicy.Constraints) == 0 && len(wi.ConsumerPolicy.Workloads) > 1 {
+                        EmptySvcPolicy := externalpolicy.ExternalPolicy{
+                                Properties:  []externalpolicy.Property{},
+                                Constraints: []string{},
+                        }
+
+                        compatible,_,_  := b.checkPolicyCompatibility(workerId, wi, nodePolicy, &wi.ConsumerPolicy, &EmptySvcPolicy, "", msgPrinter)
+                        if !compatible {
+                                // Not compatible with the constraints of the deployment policy so no need to continue checking with the service versions
+                                return
+                        }
+                }
+        }
+
 	// Generate an agreement ID
 	agreementIdString, aerr := cutil.GenerateAgreementId()
 	if aerr != nil {
 		glog.Errorf(BAWlogstring(workerId, fmt.Sprintf("error generating agreement id %v", aerr)))
 		return
 	}
-	glog.V(5).Infof(BAWlogstring(workerId, fmt.Sprintf("using AgreementId %v", agreementIdString)))
+	if glog.V(5) {glog.Infof(BAWlogstring(workerId, fmt.Sprintf("using AgreementId %v", agreementIdString)))}
```

Comment:
Just use glog.V(5).Infof

Reply:
talked to Dave... its an optimization.. will leave as added